blenderproc/python/postprocessing/PostProcessingUtility.py

```diff
@@ -275,6 +278,151 @@ def trim_redundant_channels(image: Union[list, np.ndarray]) -> Union[list, np.nd
+                        default_value = default_values[current_attribute]
+                    elif default_values and current_attribute in default_values:
+                        default_value_set = True
+                        default_value = default_values[current_attribute]
+
+                    for object_id in object_ids:
+                        # get current object
+                        current_obj = object_ids_to_object[object_id]
+
+                        # if the current obj has an attribute with that name -> get it
+                        if hasattr(current_obj, current_attribute):
+                            value = getattr(current_obj, current_attribute)
+                        # if the current object has a custom property with that name -> get it
+                        elif current_attribute in current_obj:
+                            value = current_obj[current_attribute]
+                        elif current_attribute.startswith("cf_"):
+                            if current_attribute == "cf_basename":
+                                value = current_obj.name
+                                if "." in value:
+                                    value = value[:value.rfind(".")]
+                            else:
+                                raise ValueError(f"The given attribute is a custom function: \"cf_\", but it is not "
+                                                 f"defined here: {current_attribute}")
+                        elif default_value_set:
+                            # if none of the above applies use the default value
+                            value = default_value
+                        else:
+                            # if the requested current_attribute is not a custom property or an attribute
+                            # or there is a default value stored
+                            # it throws an exception
+                            raise RuntimeError(f"The object \"{current_obj.name}\" does not have the "
+                                               f"attribute: \"{current_attribute}\". Maybe try a default value.")
+
+                        # save everything which is not instance also in the .csv
+                        if isinstance(value, (int, float, np.integer, np.floating)):
+                            found_a_non_csv_value = True
+                            resulting_map[stereo_image == object_id] = value
+                            found_dtype = type(value)
+
+                        if isinstance(value, (mathutils.Vector, mathutils.Matrix)):
+                            value = np.array(value)
+
+                        if object_id in non_image_attributes:
+                            non_image_attributes[object_id][current_attribute] = value
+                        else:
+                            non_image_attributes[object_id] = {current_attribute: value}
+
+                    if found_a_non_csv_value:
```

Comment:
`found_a_non_csv_value` and `found_dtype` are redundant, I think.

The whole segmap code is quite convoluted, but I think it was already the case before this PR, just at a different place. So maybe we could clean it later at some point.

Reply:
The issue is that this can work with list of images, even though this is never used. 

All other functions work with both.

Reply:
I see